source/extensions/filters/http/transformation/inja_transformer.h

```diff
@@ -82,15 +83,24 @@ class Extractor : Logger::Loggable<Logger::Id::filter> {
   absl::string_view extract(Http::StreamFilterCallbacks &callbacks,
                             const Http::RequestOrResponseHeaderMap &header_map,
                             GetBodyFunc &body) const;
-
+  std::string extractDestructive(Http::StreamFilterCallbacks &callbacks,
```

Comment:
why is this function different? it seems to be semantically the same as extract? should we merge them to one function?
if we do that i think we can also get rid of mode() below

Reply:
i see that they are only different by return value. but given that the string_view will be converted to string anyway in the transformation, there's no advantage in having extract return a string_view

Reply:
I posted an update here which I think should make `extract` and `extractDestructive` meaningfully distinct: 9a543cf3abd968a8068bb11823c6c1958080fd13

Specifically, this creates two maps, `extractions` and `destructive_extractions` which have different signatures, which I believe should mean that the `string_view` returned from `extract` will not be converted to `std::string` anymore.

If I'm missing something and it still will be converted to `std::string`, then just let me combine these into one